encrypt.go

```diff
@@ -189,15 +176,57 @@ func (es *encryptStream) init(version Version, sender BoxSecretKey, receivers []
 		return err
 	}
 
-	// Use the header hash to compute the MAC keys.
-	//
-	// TODO: Plumb the pre-computed shared keys above through to
-	// computeMACKeysSender.
-	es.macKeys = computeMACKeysSender(es.header.Version, sender, ephemeralKey, receivers, es.headerHash)
+	// TODO: Plumb the pre-computed shared ephemeral keys from
+	// boxPayloadKeyForReceivers through to computeMACKeysSender.
+	es.macKeys = computeMACKeysSender(es.header.Version, order, sender, ephemeralKey, receivers, es.headerHash)
 
 	return nil
 }
 
+func computeReceiverOrder(version Version, receiverCount int) []int {
+	switch version.Major {
+	case 1:
```

Comment:
I don't think randomizing the receivers will break anything in v1. Any reason not to do it for both?